torch/_tensor_docs.py

```diff
@@ -3232,12 +3232,13 @@ def callable(a, b) -> number
 
 add_docstr_all('scatter_',
                r"""
-scatter_(dim, index, src, reduce=None) -> Tensor
+scatter_(dim, index, src, value, reduce=None) -> Tensor
```

Comment:
This looks good! My only concern is that listing both `src` and `value` here might suggest that both are required positional arguments when it is actually an 'either/or' case as clarified by the description below.

@mruberry Do we have a convention for documenting cases like this where different overloads have different argument names/types for the same argument? Would `scatter_(dim, index, src/value, reduce=None) -> Tensor` be more appropriate here?

Reply:
That's a great question!

How did we even end up in this situation? 

So the current thing definitely looks like there are two required positional args. Some of our operators document different signatures separately, see max for an example (https://pytorch.org/docs/master/generated/torch.max.html?highlight=max#torch.max), but this is lame and sometimes it's hard to see all the signatures. Implementations with different signatures like this are often buggy, too. For instance, max doesn't accept an out argument unless dim or keepdim are also specified, I think.

Can we just have one positional argument here? We can support "argument aliases" if we want both names to be valid. When we do that, however, we typically only document the "primary" name and then just make a note about the alias later (or we don't even bother telling people).

Reply:
@mruberry Argument aliases make sense (for backward compatibility perhaps?) since it appears that people are calling these positional arguments with their parameter names. Could you point me to an example of how argument aliasing is supported please?

Reply:
We actually support it in two different ways. The first way is having our arg parsing remap arguments. This is what let's "axis" be used in place of "dim" today (and many developers and users are unaware this works at all). The second way is by adding a signature that makes the aliased parameter kwarg only. These signatures are identical (except for the aliased kwarg) and they just call the original signatures. That way users can pass two distinct names for the same argument. I think we did this for one of the new torch.linalg operations but I forget which, sorry, so I don't have an example ready.

Reply:
Sorry to rehash an old thread, but the `value` parameter is [part of an example](https://pytorch.org/tutorials/beginner/basics/transforms_tutorial.html#lambda-transforms) in the tutorial. Could we not resolve the issue by just adding `value` as a keyword argument in the documentation, and explaining that it can be used in place of `src`? That doesn't make any sense in normal python syntax, but at least there's some crumb for the user to follow.

Is this the result of difficulty in trying to abstract and simplify a complex system at the python level? Or in interfacing a high level language like python that doesn't support overloading/polymorphism/etc with a language that does (C++)? 

Either way, there's got to be a way to achieve this. Otherwise the docs entirely lack info about features that are non-deprecated and fully intended to be used (which is at least qualitatively different from what @mruberry is saying about not having documentation for aliases but supporting them for backwards compatibility, right?).